src/extension.ts

```diff
@@ -1003,21 +1011,31 @@ export async function activate(context: vscode.ExtensionContext) {
             const espMdfPath = idfConf.readParameter(
               "idf.espMdfPath"
             ) as string;
+
+            const pickItems = [];
+            if (espIdfPath) {
+              pickItems.push({
+                description: "ESP-IDF",
+                label: `Use current ESP-IDF (${espIdfPath})`,
+                target: espIdfPath,
+              });
+            }
+            if (espAdfPath) {
```

Comment:
since `espAdfPath ` is type casted `as string`, the value is `'undefined'` which is string and not `undefined` type and hence `if ` will be `true`